### DISCUSSION TITLE
Dzmation.hex and the magmarti.ppt via PTR: and DZLR:

### DISCUSSION THREAD

**Discussion by laffineuse** [thehighnibble/thehighnibble.github.io#118] [IMSAI8080esp replica]
I have tried to follow to the letter the instructions from the video "THN036 - New #IMSAI8080 esp desktop UI & firmware" but for the life of me I cannot get the Magenta Martini demo to run on my IMSAI8080esp.  

Here are the steps that I follow, please review and correct what I am doing wrong:
1. configure the IMSAI to 0000 0000 1000 1000
2. POWER ON and RUN
3. Drag and Drop dzmation.hex onto the IC icon in the TAPE: library
4. Power IMSAI OFF
5. load TAPE:, DZLR:, PTR:, and TTY: (turn them on)
6. POWER IMSAI ON and RUN
7. the crosshair appear on DZLR:
8. drag and drop magmarti.ppt onto PTR:, turn on PTR:
9. QUEUE PTR:
10. READ PTR:
.....NOTHING HAPPENS....

Thanks in advance for your help!

**Comment by nbreeden2** [comment]
Did you make the required HAL change so the PTR: can send the data to the TTY: ? 
This is required so the program running from the HEX ROM can receive the data from the PTR:
The ASR-33 Teletype had both the TTY: keyboard and PTR: on it. The HAL change is like using the switch on the TTY: to change its output from being driven by the keyboard to the paper tape reader.

Based on what you described I also don't see you loading and booting the IMSAI from the Dazzlemation ROM image. I believe you did based on the fact you see the cursor on the Dazzler. Does the cursor mover around as described in the video using the keyboard before the paper tape is read?

The video below describes the full process and includes how to make the HAL and ROM changes.

https://www.youtube.com/watch?v=p3aPnFFPc6w&t=7s

**Reply by laffineuse** [reply]
**Thank you** very much!  That did the trick.  I added the following two lines to boot.conf and the demo worked like a charm:
SIO1.portA.device=WEBPTR+,WEBTTY,UART0
SI02.portA.device=UART1

**Comment by nbreeden2** [comment]
Awesome, glad it works!
